pymbolic/mapper/__init__.py

```diff
@@ -446,10 +483,14 @@ def map_comparison(self, expr, *args, **kwargs):
                 self.rec(expr.right, *args, **kwargs))
 
     def map_list(self, expr, *args, **kwargs):
-        return [self.rec(child, *args, **kwargs) for child in expr]
+        return type(expr)(tuple(self.rec(child, *args, **kwargs) for child in expr))
```

Comment:
This looks worse than before, and it also doesn't implement the identical-object logic. Maybe piggyback off of `map_tuple` and just recast to a list after?

Reply:
I changed it to original way. I decided not to do idential-object logic because the list is mutable.

Reply:
Fair.